engine/src/main/java/org/kigalisim/engine/support/PopulationChangeRecalcStrategy.java

```diff
@@ -0,0 +1,100 @@
+import org.kigalisim.engine.state.Scope;
+
+/**
+ * Strategy for recalculating population changes.
+ */
+public class PopulationChangeRecalcStrategy implements RecalcStrategy {
+
+  private Scope scope;
+  private Boolean subtractRecharge;
+
+  /**
+   * Create a new PopulationChangeRecalcStrategy.
+   *
+   * @param scope The scope to use for calculations, null to use engine's current scope
+   * @param subtractRecharge Whether to subtract recharge, null to default to true
+   */
+  public PopulationChangeRecalcStrategy(Scope scope, Boolean subtractRecharge) {
+    this.scope = scope;
+    this.subtractRecharge = subtractRecharge;
+  }
+
+  @Override
+  public void execute(Engine target, RecalcKit kit) {
+    // Move the logic from SingleThreadEngine.recalcPopulationChange
+    OverridingConverterStateGetter stateGetter =
+        new OverridingConverterStateGetter(kit.getStateGetter());
+    UnitConverter unitConverter = new UnitConverter(stateGetter);
+    Scope scopeEffective = scope != null ? scope : target.getScope();
+    boolean subtractRechargeEffective = subtractRecharge != null ? subtractRecharge : true;
+    String application = scopeEffective.getApplication();
+    String substance = scopeEffective.getSubstance();
+
+    if (application == null || substance == null) {
+      ExceptionsGenerator.raiseNoAppOrSubstance("recalculating population change", "");
+    }
+
+    // Get prior population
+    EngineNumber priorPopulationRaw = target.getStream("priorEquipment", scopeEffective, null);
+    EngineNumber priorPopulation = unitConverter.convert(priorPopulationRaw, "units");
+    stateGetter.setPopulation(priorPopulation);
+
+    // Get substance sales
+    EngineNumber substanceSalesRaw = target.getStream("sales", scopeEffective, null);
+    EngineNumber substanceSales = unitConverter.convert(substanceSalesRaw, "kg");
+
+    // Get recharge volume using the calculator
+    EngineNumber rechargeVolume = RechargeVolumeCalculator.calculateRechargeVolume(
+        scopeEffective, 
+        kit.getStateGetter(), 
+        kit.getStreamKeeper(),
+        target);
+
+    // Get total volume available for new units
+    BigDecimal salesKg = substanceSales.getValue();
+    BigDecimal rechargeKg = subtractRechargeEffective ? rechargeVolume.getValue() : BigDecimal.ZERO;
+    BigDecimal availableForNewUnitsKg = salesKg.subtract(rechargeKg);
+
+    // Convert to unit delta
+    EngineNumber initialChargeRaw = target.getInitialCharge("sales");
+    EngineNumber initialCharge = unitConverter.convert(initialChargeRaw, "kg / unit");
+    BigDecimal initialChargeKgUnit = initialCharge.getValue();
+    BigDecimal deltaUnitsRaw = DivisionHelper.divideWithZero(
+        availableForNewUnitsKg, initialChargeKgUnit);
+    BigDecimal deltaUnits = deltaUnitsRaw;
+    EngineNumber newUnitsMarginal = new EngineNumber(
+        deltaUnits.compareTo(BigDecimal.ZERO) < 0 ? BigDecimal.ZERO : deltaUnits, "units");
```

Comment:
Please use line continuation style here where there is one parameter per line with a comma at the end.